pkg/runner/types.go

```diff
@@ -37,10 +37,9 @@ const (
 
 var workerAffinity = &corev1.Affinity{
 	NodeAffinity: &corev1.NodeAffinity{
-		PreferredDuringSchedulingIgnoredDuringExecution: []corev1.PreferredSchedulingTerm{
-			{
-				Weight: 100,
-				Preference: corev1.NodeSelectorTerm{
+		RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
```

Comment:
I previously had this in k8s-netperf and found it to cause more issues because the pods would never launch -- if the cluster didn't 100% match the scheduling (ie, 1 worker per az).

Reply:
This is why I have added metadata for k8s-netperf to tell the user if the pods were acrossAZ or not for example...

Reply:
Yeah, in this case it's not about AZs, but about the availability of worker nodes. With this logic, if the pods are not able to be scheduled across the "Ready" worker nodes of the cluster the test will hang.


https://github.com/cloud-bulldozer/ingress-perf/blob/db066656d18393c661585dbc9fafaa9565744569/pkg/runner/runner.go#L243-L256